Sources/SwiftParser/Lexer/Cursor.swift

```diff
@@ -242,6 +256,9 @@ extension Lexer {
 
     /// If we have already lexed a token, the kind of the previously lexed token
     var previousTokenKind: RawTokenKind?
+    
+    /// If we have already lexed a token, the `NewlinePresence` of the previously lexed token
+    var previousTokenNewlinePresence: NewlinePresence?
```

Comment:
Just a naming suggestion: I’d like to make it absolutely clear that this is about whether the previous token’s *trailing* trivia has a newline and avoid any confusion of whether the previous token had a newline inside its leading trivia. So I’d go with

```suggestion
    
    /// If we have already lexed a token, stores whether the previous lexeme‘s trailing trivia contains a newline.
    var previousLexemeTrailingTrivaNewlinePresence: NewlinePresence?
```

Reply:
Great suggestion! I will try to make the modifications along with the changes mentioned above. Thank you.